### ISSUE TITLE
Could not find matching IP for MAC address

### ISSUE THREAD

**Issue by raveious** [docker/machine#4215]
I'm not entirely sure how I managed to do this, but when making a new docker machine ( through `docker-machine create`), the boot2docker VM doesn't seem to be getting an IP address for the network interface that is associated with the displayed MAC address. I figured this out by opening VirtualBox while the machine is running, having it show me the screen of the VM, and looking at `ifconfig`.

My work around for the issue was to manually set the IP address of the VM on that interface and everything seemed to start working again, as it was before. Is there an actual fix for this that anyone is aware of?

Things to note:
- Windows 7 Pro - 64 bit
- Docker Version 17.06.0-ce
- boot2docker VM has 2 ethernet interfaces?

**Comment by raveious** [comment]
Another thing of note: Even though I get this error still. `docker-machine ssh` still works for getting into the VM without any issue...

**Comment by raveious** [comment]
Narrowed down underlying issue to the firewall configuration, blocking the VirtualBox DHCP service from issuing an IP address to the boot2docker VM.

**Comment by jkester** [comment]
This was a big pain for me too. In the end I found a tip how to create the ip for eth1 manually, add it to bootsynch script, restart default, and generate certs again. See
https://stackoverflow.com/questions/34336218/is-there-a-way-to-force-docker-machine-to-create-vm-with-a-specific-ip

Now I am stuck with next problem:
C:\Program Files\Docker Toolbox\docker.exe: error during connect: Post http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.31/containers/create: open //./pipe/docker_engine: The system cannot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.
See 'C:\Program Files\Docker Toolbox\docker.exe run --help'.

**Comment by farzadmf** [comment]
Hi @raveious , I know it's been a while here, but I would really appreciate if you elaborate a bit on what you said:

> Narrowed down underlying issue to the firewall configuration

Can you please tell me how you found out, and were you able to fix it? I'm asking this because I receive the error on my work machine but not on my home computer (on Windows).

I even tried disabling firewall completely, but it still doesn't work.

Thank you in advance

**Comment by raveious** [comment]
If I'm remembering correctly, I was able to figure it out by looking in the firewall settings and checking to see if the DHCP service for VirtualBox was able to access the virtual NIC that VirtualBox was using. The company I was working for at the time pushed an policy update that disabled a bunch of stuff right before this broke, including a firewall configuration that explicitly blocked stuff like this. After talking to some of the IT overloads, we decided that the best option was to move to Win 10 + Hyper-V because the company had to move that way anyways.
